salt/fileclient.py

```diff
@@ -1035,60 +1054,57 @@ def get_file(self,
             '\'%s\'', saltenv, dest2check, path
         )
 
+        # Compare local copy with master and skip download
+        # if no difference found.
+
         if dest2check and os.path.isfile(dest2check):
-            if not salt.utils.is_windows():
-                hash_local, stat_local = \
-                    self.hash_and_stat_file(dest2check, saltenv)
-                try:
-                    mode_local = stat_local[0]
-                except (IndexError, TypeError):
-                    mode_local = None
+            # stat local file
+
+            if support_mtime:
+                stat_local = self.stat_file(dest2check, saltenv)
             else:
-                hash_local = self.hash_file(dest2check, saltenv)
-                mode_local = None
-
-            if hash_local == hash_server:
-                if not salt.utils.is_windows():
-                    if mode_server is None:
-                        log.debug('No file mode available for \'%s\'', path)
-                    elif mode_local is None:
-                        log.debug(
-                            'No file mode available for \'%s\'',
-                            dest2check
-                        )
-                    else:
-                        if mode_server == mode_local:
-                            log.info(
-                                'Fetching file from saltenv \'%s\', '
-                                '** skipped ** latest already in cache '
-                                '\'%s\', mode up-to-date', saltenv, path
-                            )
-                        else:
-                            try:
-                                os.chmod(dest2check, mode_server)
-                                log.info(
-                                    'Fetching file from saltenv \'%s\', '
-                                    '** updated ** latest already in cache, '
-                                    '\'%s\', mode updated from %s to %s',
-                                    saltenv,
-                                    path,
-                                    salt.utils.st_mode_to_octal(mode_local),
-                                    salt.utils.st_mode_to_octal(mode_server)
-                                )
-                            except OSError as exc:
-                                log.warning(
-                                    'Failed to chmod %s: %s', dest2check, exc
-                                )
-                    # We may not have been able to check/set the mode, but we
-                    # don't want to re-download the file because of a failure
-                    # in mode checking. Return the cached path.
-                    return dest2check
-                else:
+                stat_local, hash_local = self.hash_and_stat_file(dest2check, saltenv)
```

Comment:
This runs the master func, so it gets the server's hash and stat. Also, you appear to have the arguments in the wrong order, since this function returns the hash and stat, in that order. Not the stat and hash.